### PR TITLE
fix: forest is not able to sync when kademlia is disabled

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- Connect to seed peers when `kademlia` is disabled

## Reference issue to close (if applicable)

Closes https://github.com/ChainSafe/forest/issues/3633

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: scripts/tests/calibnet_no_discovery_check.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# This script tests forest behaviours when discovery(mdns and kademlia) is disabled
+
+source "$(dirname "$0")/harness.sh"
+
+$FOREST_PATH --chain calibnet --encrypt-keystore false --mdns false --kademlia false --auto-download-snapshot --log-dir "$LOG_DIRECTORY" --detach --save-token ./admin_token
+FULLNODE_API_INFO="$(cat admin_token):/ip4/127.0.0.1/tcp/2345/http"
+export FULLNODE_API_INFO
+
+wait_util_rpc_is_ready
```
Comment (lemmih) [inline scripts/tests/calibnet_no_discovery_check.sh]: `--detach` is supposed to wait until the RPC is ready. If that isn't happening then we should fix it.

File: scripts/tests/calibnet_no_discovery_check.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+source "$(dirname "$0")/harness.sh"
+
+$FOREST_PATH --chain calibnet --encrypt-keystore false --mdns false --kademlia false --auto-download-snapshot --log-dir "$LOG_DIRECTORY" --detach --save-token ./admin_token
+FULLNODE_API_INFO="$(cat admin_token):/ip4/127.0.0.1/tcp/2345/http"
+export FULLNODE_API_INFO
+
+wait_until_rpc_is_ready
+
+# Verify that one of the seed nodes has been connected to
+until $FOREST_CLI_PATH net peers | grep "bootstrap-0.calibration.fildev.network"; do
```
Comment (LesnyRumcajs) [inline scripts/tests/calibnet_no_discovery_check.sh:15]: I guess it's unlikely, but we should still have some timeout logic.

File: scripts/tests/harness.sh [hunk 3]
```diff
@@ -80,6 +80,21 @@ function forest_print_logs_and_metrics {
   cat "$LOG_DIRECTORY"/*
 }
 
+function wait_until_rpc_is_ready {
```
Comment (LesnyRumcajs) [inline scripts/tests/harness.sh:83]: Same here. A sane timeout is always helpful.

File: scripts/tests/calibnet_no_discovery_check.sh (hunk 2 above)
Comment (hanabi1224) [reply scripts/tests/calibnet_no_discovery_check.sh:13]: Added `timeout-minutes: 10` to the workflow yaml

File: scripts/tests/harness.sh (hunk 3 above)
Comment (hanabi1224) [reply scripts/tests/harness.sh]: This has been removed

File: scripts/tests/calibnet_no_discovery_check.sh (hunk 1 above)
Comment (hanabi1224) [reply scripts/tests/calibnet_no_discovery_check.sh]: You are right, `--detach` does wait until the RPC port is listened on.

File: src/libp2p/discovery.rs [hunk 4]
```diff
@@ -179,6 +184,10 @@ pub struct DiscoveryBehaviour {
     peer_addresses: HashMap<PeerId, HashSet<Multiaddr>>,
     /// Number of connected peers to pause discovery on.
     target_peer_count: u64,
+    /// Seed peers
+    user_defined: Vec<(PeerId, Multiaddr)>,
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply src/libp2p/discovery.rs]: Fixed

**Review by lemmih — APPROVED** [review]

**Pull request merged**